src/components/AudioPlayer/RepeatAudioModal/RepeatAudioModal.tsx

```diff
@@ -0,0 +1,130 @@
+import { useMemo, useState, useEffect } from 'react';
+
+import styles from './RepeatAudioModal.module.scss';
+import RepeatSetting from './RepeatSetting';
+import SelectType, { RepeatType } from './SelectType';
+
+import Modal from 'src/components/dls/Modal/Modal';
+import Separator from 'src/components/dls/Separator/Separator';
+import { RangeVerseItem } from 'src/components/Verse/AdvancedCopy/SelectorContainer';
+import { getChapterData } from 'src/utils/chapter';
+import { generateChapterVersesKeys } from 'src/utils/verse';
+
+type RepeatAudioModalProps = {
+  chapterId: string;
+  isOpen: boolean;
+  onClose: () => void;
+  defaultRepeatType: RepeatType;
+  defaultSelectedVerse?: string;
+};
+
+const RepeatAudioModal = ({
+  chapterId,
+  isOpen,
+  onClose,
+  defaultRepeatType,
+  defaultSelectedVerse,
+}: RepeatAudioModalProps) => {
+  const chapterName = useMemo(() => {
+    const chapterData = getChapterData(chapterId);
+    return chapterData?.nameSimple;
+  }, [chapterId]);
+
+  const comboboxVersesItems = useMemo<RangeVerseItem[]>(() => {
+    const keys = generateChapterVersesKeys(chapterId);
+
+    const initialState = keys.map((chapterVersesKey) => ({
+      id: chapterVersesKey,
+      name: chapterVersesKey,
+      value: chapterVersesKey,
+      label: chapterVersesKey,
+    }));
+
+    return initialState;
+  }, [chapterId]);
+
+  const firstVerseKeyInThisChapter = comboboxVersesItems[0].value;
+  const lastVerseKeyInThisChapter = comboboxVersesItems[comboboxVersesItems.length - 1].value;
+
+  // TODO: connect to redux when the data flow is ready
+  const [repeatVerse, setRepeatVerse] = useState(() => {
+    return { total: 1, verseKey: defaultSelectedVerse || firstVerseKeyInThisChapter };
+  });
+  const [repeatVerseRange, setRepeatVerseRange] = useState({
+    total: 1,
+    from: firstVerseKeyInThisChapter,
+    to: lastVerseKeyInThisChapter,
+  });
+  const [delayBetweenVerse, setDelayBetweenVerse] = useState(0);
+
+  // reset repeatVerseRange's `to` and `from`, when chapter changed
+  useEffect(() => {
+    setRepeatVerseRange((prevRepeatVerseRange) => ({
+      ...prevRepeatVerseRange,
+      from: firstVerseKeyInThisChapter,
+      to: lastVerseKeyInThisChapter,
+    }));
+  }, [chapterId, firstVerseKeyInThisChapter, lastVerseKeyInThisChapter]);
+
+  const onClickPlay = () => {
+    onClose();
+  };
+  const onClickCancel = () => {
+    onClose();
+  };
+
+  return (
+    <Modal isOpen={isOpen} onClickOutside={onClose}>
```

Comment:
outsideClick is technically not following the convention. 

https://stackoverflow.com/questions/56051358/function-name-onbuttonclick-or-onclickbutton/56053062